src/Tools/dotnet-user-jwts/src/dotnet-user-jwts.csproj

```diff
@@ -13,13 +13,16 @@
 
   <ItemGroup>
     <Compile Include="$(SharedSourceRoot)CommandLineUtils\**\*.cs" LinkBase="Shared" />
-    <Compile Include="$(ToolSharedSourceRoot)CommandLine\**\*.cs" LinkBase="Shared" />
+    <Compile Include="$(ToolSharedSourceRoot)CommandLine\**\*.cs"
+             Exclude="$(ToolSharedSourceRoot)CommandLine\*Reporter.cs;$(ToolSharedSourceRoot)CommandLine\*Console.cs"
+             LinkBase="Shared" />
   </ItemGroup>
 
   <ItemGroup>
     <Reference Include="System.IdentityModel.Tokens.Jwt" />
     <Reference Include="Microsoft.Extensions.Configuration.Abstractions" />
     <Reference Include="Microsoft.Extensions.Configuration" />
     <Reference Include="Microsoft.Extensions.Configuration.UserSecrets" />
+    <Reference Include="dotnet-user-secrets" />
```

Comment:
No this is pretty strange. We should do shared source here.

Reply:
>  Have we already done that before?

We've done this for `dotnet-getdocument` which is a dependency of the `Extensions.ApiDescription` projects. 

> We should do shared source here.

Th challenge with doing shared source is that the resolution of user secrets depends on the MSBuild targets that are bundled as part of the tool. We could make the targets a shared source as well but I don't think we have precedence for that either and it feels yuckier than taking a dependency on the tool. Or more so, the tool's APIs.

Reply:
> > Have we already done that before?

> We've done this for `dotnet-getdocument` which is a dependency of the `Extensions.ApiDescription` projects. 

This isn't a valid analogy because Microsoft.Extensions.ApiDescription.Server contains both an MSBuild `Task` and the output of the `dotnet-getdocument` project. There's no `dotnet-getdocument` package for the .Server package to reference.

Reply:
Are you referring to this?

https://github.com/dotnet/aspnetcore/blob/c8a3c224e7e55e04583eee7559452d74e0358382/src/Tools/Extensions.ApiDescription.Server/src/Microsoft.Extensions.ApiDescription.Server.csproj#L17-L26

That seems a little different since it's a development dependency rather than a dotnet tool and it internalizes the entire reference and copies dotnet-getdocument.dll/runtimconfig.json/depts.json into its own package. And it's just invoking the tool like a command line utility rather than loading the assembly:

https://github.com/dotnet/aspnetcore/blob/c53700bf8a265303f90e0b1f73be4f27c756f6b7/src/Tools/Extensions.ApiDescription.Server/src/build/Microsoft.Extensions.ApiDescription.Server.targets#L54

I don't think we're doing this for `dotnet-user-secrets` either:

https://github.com/dotnet/aspnetcore/blob/0ee742c53f2669fd7233df6da89db5e8ab944585/eng/ProjectReferences.props#L58

> We could make the targets a shared source as well but I don't think we have precedence for that either and it feels yuckier than taking a dependency on the tool.

I still think this is probably cleaner and easier to reason about.

Reply:
What's the best way to factor this out while accounting for the fact that the implementation here depends on a build target?

Reply:
For the target, I bet you can do something like the following and then just do shared source the normal way.

```
<None Include="..\..\dotnet-user-secrets\src\assets\**\*" CopyToOutputDirectory="PreserveNewest" CopyToPublishDirectory="PreserveNewest" />
```

Reply:
It's probably better to put the target and the shared source files in an actual shared location relative to the solution root rather than navigating the relative path to the `dotnet-user-secrets` project though.

Reply:
OK, I refactored this to a shared source model. Still feels weird to interact with targets this way but it does the job...